lib/srv/app/azure/handler_test.go

```diff
@@ -74,8 +74,13 @@ func TestForwarder_getToken(t *testing.T) {
 						}
 					}
 
+					// advance time by getTokenTimeout
 					clock.Advance(getTokenTimeout)
+					// block for 2*getTokenTimeout; effectively forever, since clock won't advance more than getTokenTimeout.
```

Comment:
Does this also mean this go routine will stay forever?

Reply:
No: the parent goroutine is called with the context, which gets canceled when the timeout occurs.

Reply:
The parent goroutine (FnCache.get) gets cancelled and exits. But the goroutine that runs this `getAccessToken` never exits as `clock.Sleep` doesn't stop on the `ctx`?

Reply:
Right, I see what you mean. I think this is a minor issue as the tests are naturally short-lived, but we can clean those up by unblocking the call after the test is done.

Reply:
Ya, it's not a big deal in this case